src/NrfTerminalCommander.ts

```diff
@@ -215,7 +213,7 @@ export default class NrfTerminalCommander implements ITerminalAddon {
         }
 
         if (terminalMode.type == 'character') {
-            this.#keyListener = this.#terminal.onKey(terminalMode.onKey);
+            // We don't need to set onKey to achieve character mode
```

Comment:
I believe you must set onKey to a hardcoded no-op here, otherwise the previously set onKey from the line mode is still active.

Reply:
In the lines above we do:
```
if (this.#keyListener !== null) {
            this.#keyListener.dispose();
            this.#keyListener = null;
}
```
This should clean up the last assigned `onKey` handler. I think you can register as many as you like (the other addons add some of their own) so it is a case of tracking what was last added and removing it which is what that if-statement does.

Reply:
Sorry about my ignorance :)